src/preamble.js

```diff
@@ -361,7 +361,7 @@ function allocate(slab, types, allocator, ptr) {
 function getMemory(size) {
   if (!staticSealed) return staticAlloc(size);
   if (!runtimeInitialized) return dynamicAlloc(size);
-  return _malloc(size);
+  return _calloc(1, size);
```

Comment:
A downside to doing this is that it means getMemory is slower for all other users. I think we should instead find the place calling getMemory here and zero out the memory afterwards.

Another issue is the use of calloc - we can't depend on it always existing (unlike malloc which we do link in by default). We could ensure it is linked in when necessary, but zeroing out the memory afterwards is probably simpler and good enough here.

Reply:
Sure.  The place that matters is where the static memory for a SIDE_MODULE is allocated -- the other part of this patch in `jsifier.js`.  I'll update this to fix it there.

Reply:
I'm having a little trouble turning what I know needs to happen into a patch.

Basically, the allocation of the static data area in a SIDE_MODULE is:

```
gb = alignMemory(getMemory(205120), 8 || 1);
```
```
/* memory initializer */ allocate([0,0,0,0,0,0,240,63...], "i8", ALLOC_NONE, gb);
```

where the amount of memory allocated is more than the memory explicitly initialized.  The additional memory at the end is supposed to be zero'd, but it isn't being zero'd, so it should just be a matter of writing a loop from the end of the explicitly memory initializer to the end of the static block.  However, these two lines are written in different parts of the code (`jsifier.js` and fastcomp's `JSBackend.cpp`) and I don't think either place has both pieces of information necessary to write this loop.  Any thoughts?

Reply:
Hmm, how about doing the zeroing out right after the first line? You know `gb` and the size `205120` there. It's not maximally efficient I suppose, but dynamic linking will be overhauled eventually anyhow - can just leave a comment there about the overhead.